rpc/handler.go

```diff
@@ -388,7 +406,23 @@ func (h *handler) handleCallMsg(ctx *callProc, msg *jsonrpcMessage, stream *json
 		}
 		return nil
 	case msg.isCall():
+		if h.rpcSlowLogThreshold > 0 && h.isRpcMethodNeedsCheck(msg.Method) {
+			slowTimer := time.AfterFunc(time.Duration(h.rpcSlowLogThreshold)*time.Millisecond, func() {
+				h.logger.Warn("[rpc.slow] - running", "method", msg.Method)
+			})
+
+			defer slowTimer.Stop()
+		}
+
 		resp := h.handleCall(ctx, msg, stream)
+
+		if h.rpcSlowLogThreshold > 0 && h.isRpcMethodNeedsCheck(msg.Method) {
```

Comment:
let's store result of `h.isRpcMethodNeedsCheck(msg.Method)` into local bool variable